File: LoanApplicationWebAPI/Models/LoanApplication.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace LoanApplicationWebAPI.Models
{
    public class LoanApplication
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int LoanApplicationId { get; set; }

        [Required]
        public LoanType LoanType { get; set; }

        [Required]
        [Range(1000, double.MaxValue)]
        [Column(TypeName = "decimal(18,2)")]
        public decimal LoanAmount { get; set; }

        //[Required]
        //[Column(TypeName = "decimal(5,2)")]
        //public decimal InterestRate { get; set; } = 0;

        public ApplicationStatus Status { get; set; }= ApplicationStatus.Pending;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Foreign Keys
        public int CustomerId { get; set; }
        public Customer? Customer { get; set; }

        public int LoanOfficerId { get; set; }
        public LoanOfficer? LoanOfficer { get; set; }

        public int LoanSchemeId { get; set; }
        public LoanScheme? LoanScheme { get; set; }

        [JsonIgnore]
        public LoanApproved? LoanApproved { get; set; }
    }
}

File: src/app/Models/loan-application.ts
import { Customer } from './customer';
import { LoanOfficer } from './loan-officer';
import { LoanScheme } from './loan-scheme';

export interface LoanApplication {
  loanApplicationId: number;
  loanType: string;
  loanAmount: number;
  status: 'Pending' | 'Approved' | 'Rejected';
  createdAt: Date;
  customerId: number;
  customer?: Customer;
  loanOfficerId?: number;
  loanOfficer?: LoanOfficer;
  loanSchemeId: number;
  loanScheme?: LoanScheme;
}
